typing/typecore.ml

```diff
@@ -3238,7 +3238,12 @@ and type_expect_
       re { exp with exp_extra =
              (Texp_poly cty, loc, sexp.pexp_attributes) :: exp.exp_extra }
   | Pexp_newtype({txt=name}, sbody) ->
-      let ty = newvar () in
+      let ty =
+        if Typetexp.valid_tyvar_name name then
+          newvar ()
+        else
+          newvar ~name ()
+      in
```

Comment:
It looks like it is the other way round. Shouldn't you use `name` only if it is valid?

Reply:
This seems to be the reason the CI fails.

Reply:
Woops, sorry! Should be fixed now.